File: spring/fizz-buzz/src/main/java/de/welcz/fizzbuzz/FizzBuzzController.java
package de.welcz.fizzbuzz;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/v1/fizz-buzz/")
public class FizzBuzzController {
    private final FizzBuzzService service;

    @Autowired
    public FizzBuzzController(FizzBuzzService service) {
        this.service = service;
    }

    @GetMapping("/numbers/{input}")
    public String getNumber(@PathVariable int input) {
        return service.calculate(input);
    }
}

File: kotlin/spring/src/main/kotlin/fizz/buzz/FizzBuzzController.kt
package fizz.buzz

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
class FizzBuzzController @Autowired
constructor(private val fizzBuzz: FizzBuzzService) {
    @ResponseStatus(HttpStatus.BAD_REQUEST, reason = "invalid parameters")
    @ExceptionHandler(IllegalArgumentException::class)
    fun handleIllegalArguments() {
    }
    
    @RequestMapping("/fizz-buzz/numbers/{limit}")
    fun getNumbers(@PathVariable("limit") limit: Int): Array<String> {
        require(limit in 1..1000) { "limit must be between 1 and 1000" }
        return fizzBuzz.calculateUpTo(limit)
    }
}
